tools/c7n_azure/c7n_azure/session.py

```diff
@@ -169,19 +172,54 @@ def load_auth_file(self, path):
                 tenant=data['credentials']['tenant']
             ), data['subscription'])
 
-    def get_auth_string(self):
-        if type(self.credentials) is not ServicePrincipalCredentials:
+    def get_functions_auth_string(self):
+        """
+        Build auth json string for deploying
+        Azure Functions.  Look for dedicated
+        Functions environment variables or
+        fall back to normal Service Principal
+        variables.
+
+        """
+
+        function_auth_variables = [
+            'AZURE_FUNCTION_TENANT_ID',
+            'AZURE_FUNCTION_CLIENT_ID',
+            'AZURE_FUNCTION_CLIENT_SECRET'
+        ]
+
+        # Check for override on subscription ID
+        if 'AZURE_FUNCTION_SUBSCRIPTION_ID' in os.environ:
+            function_subscription_id = os.environ['AZURE_FUNCTION_SUBSCRIPTION_ID']
```

Comment:
nit: function_subscription_id = os.environ.get('AZURE_FUNCTION_SUBSCRIPTION_ID', self.subscription_id) ?

Reply:
Ugh, thank you.  Been away from Python for 3 months and now its like learning to ride a bike again....